### PR TITLE
Implement pool for high prio requests based on db events

### PR DESCRIPTION
**Pull request by kroesctrl**
Implement pool for high prio requests based on db events

### CODE REVIEW

File: osgp/platform/osgp-throttling-service/src/main/java/org/opensmartgridplatform/throttling/PermitsPerNetworkSegment.java [hunk 1]
```diff
@@ -134,9 +131,74 @@ public boolean releasePermit(
...
             throttlingConfigId, clientId, baseTransceiverStationId, cellId, requestId);
 
+    if (this.useHighPrioPool()) {
+      this.permitReleasedNotifier.notifyPermitReleased(baseTransceiverStationId, cellId);
+    }
+
     return numberOfReleasedPermits == 1;
   }
 
+  private boolean useHighPrioPool() {
```
Comment (harrymiddelburgetall) [inline osgp/platform/osgp-throttling-service/src/main/java/org/opensmartgridplatform/throttling/PermitsPerNetworkSegment.java]: Can't we use an explicite use boolean in the properties here?

File: osgp/platform/osgp-throttling-service/src/main/java/org/opensmartgridplatform/throttling/service/PermitReleasedNotifier.java [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+  private void closeConnection(
+      final Connection connection, final int baseTransceiverStationId, final int cellId) {
+    try {
+      if (connection != null) {
+        connection.close();
+      }
+    } catch (final SQLException e) {
+      log.error(
+          "SQLException occurred while listening for notification for btsId: {}, cellId: {}",
```
Comment (harrymiddelburgetall) [inline osgp/platform/osgp-throttling-service/src/main/java/org/opensmartgridplatform/throttling/service/PermitReleasedNotifier.java]: SQLException occurred while **trying to close the connection that is** listening for notification for btsId: {}, cellId: {}

**Review by harrymiddelburgetall — CHANGES_REQUESTED** [review]
two minor remarks

File: osgp/platform/osgp-throttling-service/src/main/java/org/opensmartgridplatform/throttling/service/PermitReleasedNotifier.java (hunk 2 above)
Comment (kroesctrl) [reply osgp/platform/osgp-throttling-service/src/main/java/org/opensmartgridplatform/throttling/service/PermitReleasedNotifier.java]: Changed

File: osgp/platform/osgp-throttling-service/src/main/java/org/opensmartgridplatform/throttling/PermitsPerNetworkSegment.java (hunk 1 above)
Comment (kroesctrl) [reply osgp/platform/osgp-throttling-service/src/main/java/org/opensmartgridplatform/throttling/PermitsPerNetworkSegment.java]: Added explicit property

**Review by harrymiddelburgetall — APPROVED** [review]

**Pull request merged**